pkg/cloud-provider/cloudapi/azure/azure_security.go

```diff
@@ -576,7 +573,7 @@ func (computeCfg *computeServiceConfig) processAndBuildATSgView(networkInterface
 		}
 		vnetIDLowerCase := strings.ToLower(*networkInterface.VnetID)
 		nsgIDToVnetIDMap[nsgIDLowerCase] = vnetIDLowerCase
-		if strings.Compare(sgName, strings.ToLower(appliedToSecurityGroupNamePerVnet)) == 0 {
+		if strings.Contains(strings.ToLower(sgName), appliedToSecurityGroupNamePerVnet) {
```

Comment:
Can you create 2 VNET and verify if things look Ok there too.

Reply:
Yes, I have done it. Both the VNETs had an ANP and I could see sync happening on both.
Logically speaking, this code is just checking if the NSG has specific format. A few lines above this code, we get NSGs associated with network interface of managed VPCs. So, at this block every NSG is checked against the condition and goes inside the if condition.